### PR TITLE
WIP Ports: zsh

### PR DESCRIPTION
**Pull request by mrkmg**
This is a _working_ port of zsh. I have not tested everything yet though.

Requires #6144

### CODE REVIEW

File: Ports/.port_include.sh [hunk 1]
```diff
@@ -169,6 +169,9 @@ func_defined install || install() {
 func_defined post_install || post_install() {
     echo
 }
+func_defined post_uninstll || post_uninstll() {
+  :
+}
```
Comment (linusg) [inline Ports/.port_include.sh:174]:
```suggestion
func_defined post_uninstall || post_uninstall() {
    :
}
```

File: Ports/.port_include.sh [hunk 2]
```diff
@@ -236,6 +239,7 @@ uninstall() {
             # Without || true, mv will not be executed if you are uninstalling your only remaining port.
             grep -v "^manual $port " "$packagesdb" > packages.db.tmp || true
             mv packages.db.tmp "$packagesdb"
+            post_uninstll
```
Comment (linusg) [inline Ports/.port_include.sh:242]:
```suggestion
            post_uninstall
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: Ports/.port_include.sh (hunk 1 above)
Comment (mrkmg) [reply Ports/.port_include.sh:174]: omgosh. I can not believe I typod that 🤦‍♂️

**Comment by IdanHo** [comment]
this needs to be rebased to fix the merge conflicts and the fixup commit needs to be squashed as per the [contribution guidelines](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md)

**Pull request closed without merging**